skiko/src/jvmMain/cpp/common/awt_jni.cc

```diff
@@ -36,10 +36,10 @@ extern "C"
         awt->FreeDrawingSurface(ds);
     }
 
-    JNIEXPORT void JNICALL Java_org_jetbrains_skiko_AWTKt_lockDrawingSurface(JNIEnv *env, jobject obj, jlong drawingSurfacePtr)
+    JNIEXPORT jint JNICALL Java_org_jetbrains_skiko_AWTKt_lockDrawingSurface(JNIEnv *env, jobject obj, jlong drawingSurfacePtr)
     {
         JAWT_DrawingSurface *ds = fromJavaPointer<JAWT_DrawingSurface *>(drawingSurfacePtr);
-        ds->Lock(ds);
+        return ds->Lock(ds) & JAWT_LOCK_ERROR;
```

Comment:
`return (ds->Lock(ds) & JAWT_LOCK_ERROR) != 0;`, previous version was subtly buggy is some compilers.